File: task-03/C/subtask-03.c
#include <stdio.h>

int main() {
    int n;

    printf("Enter odd number: "); //since pattern contains only odd number of *
    scanf("%d", &n);

    if (n % 2 == 0) {
        printf("The number must be odd.\n");
        return 1;
    }


    for (int i = 1; i <= n; i += 2) {
        for (int j = 0; j < (n - i) / 2; j++)
            printf(" ");
        for (int j = 0; j < i; j++)
            printf("*");
        printf("\n");
    }
    for (int i = n - 2; i > 0; i -= 2) {
        for (int j = 0; j < (n - i) / 2; j++)
            printf(" ");
        for (int j = 0; j < i; j++)
            printf("*");
        printf("\n");
    }

    return 0;
}

File: task-03/Go/subtask-03.go
package main

import (
    "fmt"
)

func main() {
    var n int
    fmt.Print("Enter a number: ")
    fmt.Scan(&n)

    if n%2 == 0 {
        n++                  //making it the next odd
    }

    mid := (n + 1) / 2

    for i := 1; i <= mid; i++ {
        for j := 1; j <= mid-i; j++ {
            fmt.Print(" ")
        }
        for j := 1; j <= 2*i-1; j++ {
            fmt.Print("*")
        }
        fmt.Println()
    }

    for i := mid - 1; i >= 1; i-- {
        for j := 1; j <= mid-i; j++ {
            fmt.Print(" ")
        }
        for j := 1; j <= 2*i-1; j++ {
            fmt.Print("*")
        }
        fmt.Println()
    }
}
